lib/ui/painting/picture.cc

```diff
@@ -84,52 +84,43 @@ Dart_Handle Picture::RasterizeToImage(sk_sp<SkPicture> picture,
 
   auto picture_bounds = SkISize::Make(width, height);
 
-  auto ui_task = fml::MakeCopyable(
-      [ui_task_runner, image_callback = std::move(image_callback),
-       unref_queue](sk_sp<SkImage> raster_image) mutable {
-        // Send the raster image back to the UI thread for submission to the
-        // framework.
-        fml::TaskRunner::RunNowOrPostTask(
-            ui_task_runner,
-            fml::MakeCopyable([raster_image,
-                               image_callback = std::move(image_callback),
-                               unref_queue]() mutable {
-              auto dart_state = image_callback->dart_state().lock();
-              if (!dart_state) {
-                // The root isolate could have died in the meantime.
-                return;
-              }
-              tonic::DartState::Scope scope(dart_state);
-
-              if (!raster_image) {
-                tonic::DartInvoke(image_callback->Get(), {Dart_Null()});
-                return;
-              }
-
-              auto dart_image = CanvasImage::Create();
-              dart_image->set_image(
-                  {std::move(raster_image), std::move(unref_queue)});
-              auto* raw_dart_image = tonic::ToDart(std::move(dart_image));
-
-              // All done!
-              tonic::DartInvoke(image_callback->Get(), {raw_dart_image});
```

Comment:
I'd expect that the `std::unique_ptr` to be destructed here after that `image_callback` is no longer used, which means destruction in the ui thread. Is it not the case? @chinmaygarde

Reply:
In the old implementation, the `std::unique_ptr` was being captured into a ref counted lambda by `MakeCopyable`.

The `MakeCopyable` lambda was then being passed through the GPU thread to the UI thread.  Either the GPU thread or the UI thread could drop the last reference to the lambda.  If the GPU thread destroyed the lambda, then the captured image_callback would be destructed on the GPU thread, causing a failure.